File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n = nums.size();
        int index = 0;
        vector<int> res(n);
        for(int i = 0; i < n;++i ){
            if(nums[i] != 0){
                res[index++] = nums[i];
            }
        }
        for(int i = index; i<n ; ++i){
            res[i] = 0;
        }
        for(int i = 0; i < n; ++i){
            nums[i] = res[i];
        }
    }
};

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        //0th position finder
        int w = 0;
        for(int i = 0; i<nums.length;++i){
            //checking for non zero ele
            if(nums[i] != 0){
                //assignings temp to store the pos of 0th 
                int temp = nums[w];
                //write 0 to non zero element place
                nums[w] = nums[i];
                //write the non zero element to 0th place
                nums[i] = temp;
                w++;
            }
        }
    }
}